ros2/serow_ros2/src/serow_driver.cpp

```diff
@@ -72,46 +67,44 @@ class SerowDriver : public rclcpp::Node {
                 imu_measurement.angular_velocity =
                     Eigen::Vector3d(imu_data.angular_velocity.x, imu_data.angular_velocity.y,
                                     imu_data.angular_velocity.z);
-                base_imu_data_.pop();
 
-                // Create the leg F/T measurement
                 std::map<std::string, serow::ForceTorqueMeasurement> ft_measurements;
-                // Need a better way to check that all end-effectors have a new F/T
-                for (auto& [key, value] : ft_data_) {
-                    if (value.size()) {
+
+                if (ft_data_.size() == force_torque_state_subscriptions_.size()){
```

Comment:
imagine you have a biped what if F/T from leg 1 is not working but F/T from leg 2 is working then when you gather 2 F/T from leg 2 this check will be true.